sphinx/util/typing.py

```diff
@@ -147,7 +149,63 @@ def is_system_TypeVar(typ: Any) -> bool:
     return modname == 'typing' and isinstance(typ, TypeVar)
 
 
-def restify(cls: type | None, mode: str = 'fully-qualified-except-typing') -> str:
+class RenderMode(enum.Flag):
+    """Additional flags for rendering annotations or reST content."""
+
+    # principal modes (mutually exclusive with each other)
+
+    smart = enum.auto()
+    """Show the annotation name."""
+
+    fully_qualified = enum.auto()
+    """Show the module name and qualified name of the annotation.
+
+    This mode is mutually exclusive with :attr:`smart` and :attr:`fully_qualified_except_typing`.
+    """  # NoQA: E501
+
+    fully_qualified_except_typing = enum.auto()
+    """Same as :attr:`fully_qualified` but do not show module name for ``typing`` members.
+
+    This mode is mutually exclusive with :attr:`smart` and :attr:`fully_qualified`.
+    """
+
+    # secondary modes (can be used as flags on the primary modes)
+
+    short_literal = enum.auto()
+    """Use PEP 604 style to render literals."""
+
+
+_MUTUALLY_EXCLUSIVE_MODES = list(itertools.combinations([
+    RenderMode.smart,
+    RenderMode.fully_qualified,
+    RenderMode.fully_qualified_except_typing,
+], 2))
+
+
+def _normalize_mode(mode: str | RenderMode) -> RenderMode:  # for backwards compatibility
```

Comment:
Could you add a short docstring to this method?

> Before the introduction of RenderMode as an ``enum.Flag`` type, render mode was specified using string literals.  This method converts callsites that use the legacy string format into the corresponding ``Flag`` mode.

(or something better along those lines)

Reply:
Honestly, I'm not really for documenting private API since the name is self-explanatory IMO (especially with the type annotations).